File: robot_examples/go/rest_api/swagger/RobotStatus.go
package swagger

import (
)

type RobotStatus struct {
    Name  string  `json:"name,omitempty"`
    Token  string  `json:"token,omitempty"`
    Configuration  RobotConfiguration  `json:"configuration,omitempty"`
    SimulationTime  float32  `json:"simulationTime,omitempty"`
    TimeTick  float32  `json:"timeTick,omitempty"`
    RealTimeTick  float32  `json:"realTimeTick,omitempty"`
    Points  float32  `json:"points,omitempty"`
    Health  float32  `json:"health,omitempty"`
    IsDead  bool  `json:"isDead,omitempty"`
    Direction  float32  `json:"direction,omitempty"`
    Speed  float32  `json:"speed,omitempty"`
    PosX  float32  `json:"posX,omitempty"`
    PosY  float32  `json:"posY,omitempty"`
    MaxBoardX  float32  `json:"maxBoardX,omitempty"`
    MaxBoardY  float32  `json:"maxBoardY,omitempty"`
    CannonReloadingTime  float32  `json:"cannonReloadingTime,omitempty"`
    FiredNewMissile  bool  `json:"firedNewMissile,omitempty"`
    ScanStatus  ScanStatus  `json:"scanStatus,omitempty"`
    
}

File: robot_examples/haskell-servant/rest_api/lib/Model/RobotStatus.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.RobotStatus
    ( RobotStatus (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.RobotConfiguration
import Model.ScanStatus


data RobotStatus = RobotStatus
    { name :: String
    , token :: String
    , configuration :: RobotConfiguration
    , simulationTime :: Float
    , timeTick :: Float
    , realTimeTick :: Float
    , points :: Float
    , health :: Float
    , isDead :: Bool
    , direction :: Float
    , speed :: Float
    , posX :: Float
    , posY :: Float
    , maxBoardX :: Float
    , maxBoardY :: Float
    , cannonReloadingTime :: Float
    , firedNewMissile :: Bool
    , scanStatus :: ScanStatus
    } deriving (Show, Eq, Generic)

instance FromJSON RobotStatus
instance ToJSON RobotStatus
instance Arbitrary RobotStatus where
    arbitrary = RobotStatus <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
